src/Maybe.gren

```diff
@@ -74,6 +108,23 @@ withDefault default maybe =
             default
 
 
+{-| Same as [withDefault](#withDefault) but the default value is wrapped in
+a function. This is useful when computing the default value is expensive, as
+you can compute it only when it is required.
+
+In most cases you should use [withDefault](#withDefault) instead.
```

Comment:
I think you should suggest using pattern matching as well (maybe before with default)